docs/source/custom_sockets.rst

```diff
@@ -0,0 +1,214 @@
+Custom Sockets in Syncano
+=========================
+
+``Syncano`` provides possibility of creating the custom sockets. It means that there's a possibility
+to define a very specific endpoints in syncano application and use them as normal api calls.
+Currently custom sockets allow only one dependency - script. This mean that on the backend side
+each time the api is called - the script is executed and result from this script is returned as a result of the
+api call.
+
+Creating a custom socket
+------------------------
+
+There are two methods of creating the custom socket. First: use the helpers objects defined in Python Libray.
+Second: use the raw format - this is described below.
+
+To create a custom socket follow the steps::
+
+    import syncano
+    from syncano.models import CustomSocket, Endpoint, ScriptCall, ScriptDependency, RuntimeChoices
+    from syncano.connection import Connection
+
+    custom_socket = CustomSocket(name='my_custom_socket')  # this will create an object in place (do api call)
+
+    # define endpoints
+    my_endpoint = Endpoint(name='my_endpoint')  # again - no api call here
+    my_endpoint.add_call(ScriptCall(name='custom_script'), methods=['GET'])
+    my_endpoint.add_call(ScriptCall(name='another_custom_script'), methods=['POST'])
+
+    # explanation for the above lines:
+    # The endpoint will be seen under `my_endpoint` name:
+    # On this syncano api endpoint the above endpoint will be called (after custom socket creation)
+    # <host>://<api_version>/instances/<instance_name>/endpoints/sockets/my_endpoint/
+    # On this syncano api endpoint the details of the defined endpoint will be returned
+    # <host>://<api_version>/instances/<instance_name>/sockets/my_custom_socket/endpoints/my_endpoint/
+    # For the above endpoint - the two calls are defined, one uses GET method - the custom_script will be executed
+    # there, second uses the POST method and then the another_custom_script will be called;
+    # Currently only script are available for calls;
+
+    # After the creation of the endpoint, add them to custom_socket:
+    custom_socket.add_endpoint(my_endpoint)
+
+    # define dependency now;
+    # using a new script - defining new source code;
+    custom_socket.add_dependency(
+        ScriptDependency(
+            Script(
+                label='custom_script',
+                runtime_name=RuntimeChoices.PYTHON_V5_0,
+                source='print("custom_script")'
+            )
+        )
+    )
+    # using a existing script:
+    another_custom_script = Script.please.get(id=2)
+    custom_socket.add_dependency(
+        ScriptDependency(
+            another_custom_script
+        )
+    )
+
+    # now it is time to publish custom_socket;
+    custom_socket.publish()  # this will do an api call and will create script;
+
+Some time is needed to setup the environment for this custom socket.
+There is possibility to check the custom socket status::
+
+    print(custom_socket.status)
```

Comment:
it will always be "checking" after creation/update so maybe here first .reload() then print